sopt30th_1st HW/sopt30th_1st HW/LoginViewController.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  LoginViewController.swift
+//  sopt30th_1st HW
+//
+//  Created by 최영린 on 2022/04/06.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController {
+
+    @IBOutlet weak var idTextField: UITextField!
+    @IBOutlet weak var pwTextField: UITextField!
+    @IBOutlet weak var loginButton: UIButton! {
+        didSet{
+            loginButton.isEnabled = false
+        }
+    }
+    @IBOutlet weak var passwordButton: UIButton!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setBackButton()
+        setLoginButton()
+    }
+    
+    @IBAction func passwordButton(_ sender: Any) {
+        if(pwTextField.isSecureTextEntry){
+            pwTextField.isSecureTextEntry = false
+            passwordButton.setImage(UIImage(named: "PasswordHidden"), for: .normal)
+            }
+        else{
+            pwTextField.isSecureTextEntry = true
+            passwordButton.setImage(UIImage(named: "PasswordShown"), for: .normal)
+        }
+    }
+    
+    @IBAction func loginButtonDidTap(_ sender: Any) {
+        guard let welcomeVC = self.storyboard?.instantiateViewController(withIdentifier: "WelcomeViewController") as? WelcomeViewController else { return }
+        
+        welcomeVC.user = idTextField.text
+        welcomeVC.modalTransitionStyle = .crossDissolve
+        welcomeVC.modalPresentationStyle = .fullScreen
+
+        self.present(welcomeVC, animated: true, completion: nil)
+    }
+    
+    @IBAction func registerButtonDidTap(_ sender: Any) {
+        guard let registerIDVC = self.storyboard?.instantiateViewController(withIdentifier: "RegisterIDViewController") as? RegisterIDViewController else { return }
+        
+        self.navigationController?.pushViewController(registerIDVC, animated: true)
+    }
+    
+    private func setBackButton() {
+        let backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+                navigationItem.backBarButtonItem = backBarButtonItem
+    }
+    
+    private func textHandler(_ a: UIAction) -> Void {
+        if self.idTextField.text?.isEmpty == true || self.pwTextField.text?.isEmpty == true{
+            self.loginButton.isEnabled = false
+        } else {
+            self.loginButton.isEnabled = true
+        }
+    }
+    
+    private func setLoginButton() {
+        self.idTextField.addAction(UIAction(handler: self.textHandler), for: .editingChanged)
+        self.pwTextField.addAction(UIAction(handler: self.textHandler), for: .editingChanged)
+    }
```

Comment:
두개라서 두줄로도 충분히 깔끔히 가능하지만
고냥 문법 하나 알려드리구 싶어서,, 굳이 남겨봅니다~!
forEach라는 아이가 있답니다~!

```suggestion
    private func setLoginButton() {
        [idTextField, pwTextField].forEach {
        $0.addAction(UIAction(handler: self.textHandler), for: .editingChanged)
        }
    }
```

한 화면에, 이메일,비밀번호 등 TextField가 여러개면 대부분
위치만 다를 뿐 색이나 radius 값 등은 같으니까, 이를 하나하나 설정해주다보면 무의미하게 반복되는 코드가
너무 길어지다보니, 이럴 때 사용하면 좋을 듯 합니다 ..!

상황에 따라 사용하면 좋을 것 같아요! 만약 네개 다섯개 막 이렇게 되어버리면,, 너무 귀찮으니까!!

또, IBOutlet같은 경우에 IBOutlet Collection이라는 것이 있는데 알아보셔도 좋을 것 같아요!
이거 역시,, 모르시겠다면 언제든 바로 알려드리겠습니다 >__<

Reply:
짜면서도 정말 저 부분은 바꿀수 있을 것 같은데 방법을 몰라서 답답했는데 알려주셔서 정말감사해요!!!

Reply:
저도 덕분에 알아갑니다!!! 정말 저 방법 궁금했는데 forEach를 사용하는 거군요...!!